packages/components/src/modal/frame.js

```diff
@@ -70,6 +70,10 @@ class ModalFrame extends Component {
 		if ( event.keyCode === ESCAPE ) {
 			this.handleEscapeKeyDown( event );
 		}
+
+		if ( this.props.onKeyDown ) {
```

Comment:
Should we add this new prop to the modal's readme and update the changelog?

Reply:
Yeah, I'm not sure about this prop. It is needed, but I might make it experiemental so that there's the option of removing it in the future. We seem to have this issue regularly where we need to stop propagation of keyboard events from popovers and modals, and it'd be good to come up with a better pattern.

Reply:
Changed it now so that this is handled internally modal frame using a prop `__experimentalPreventKeyboardEventPropagation`.